### PR TITLE
Small cleanup in Libmscore

### PR DESCRIPTION
**Pull request by cbjeukendrup**
- Cleanup some #includes and forward declarations
- Use Color::black and Color::transparent etc. throughout the project

I would like to propose that we keep `#include`s sorted alphabetically per group (groups like Qt, libmscore, non-libmscore, log.h, etc.). It is a very small effort to put a new include at the correct line, and this makes maintenance and cleanup much easier. I even found a case where one header was included multiple times in one cpp file.

### CODE REVIEW

File: src/engraving/libmscore/textlinebase.h [hunk 1]
```diff
@@ -76,16 +76,20 @@ enum class HookType : char {
     NONE, HOOK_90, HOOK_45, HOOK_90T
 };
 
+//---------------------------------------------------------
+//   PlaceText
+//---------------------------------------------------------
+
+enum class PlaceText : char {
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/textlinebase.h:83]: I wonder whether we'd need a `Q_DECLARE_METATYPE(Ms::PlaceText);` too?

File: src/engraving/libmscore/scorefile.cpp [hunk 2]
```diff
@@ -395,13 +395,13 @@ std::shared_ptr<mu::draw::Pixmap> Score::createThumbnail()
 
     Page* page = pages().at(0);
     RectF fr  = page->abbox();
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/scorefile.cpp:397]: if reducing whitespace, then here too

File: src/engraving/draw/color.cpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ static void insertHexComponent(int num, std::stringstream& ss);
 
 const Color Color::black { 0, 0, 0 };
 const Color Color::white { 255, 255, 255 };
-const Color Color::transparent { 0, 0, 0, 0 };
+const Color Color::transparent { 255, 255, 255, 0 };
```
Comment (Jojo-Schmitz) [inline src/engraving/draw/color.cpp:21]: It seems this change, from transparent black to transparente white, breaks the mtests?

Comment (Jojo-Schmitz) [reply src/engraving/draw/color.cpp:21]: See https://github.com/musescore/MuseScore/pull/8928/checks?check_run_id=3420975345#step:8:787

Comment (Jojo-Schmitz) [reply src/engraving/draw/color.cpp:21]: strangly only for tuplets though?

Comment (Jojo-Schmitz) [reply src/engraving/draw/color.cpp:21]: No, it seems to nt the the `Color::black` lacking the trailing `, 255`? Still, why only for tuplets?
It now wants `<color r="0" g="0" b="0" a="255"/>` being set explicitly.

Comment (cbjeukendrup) [reply src/engraving/draw/color.cpp:21]: But even if I add it, the tests are still failing...

Comment (Jojo-Schmitz) [reply src/engraving/draw/color.cpp:21]: Yeah, something very strange is goind on there. if it were the missinf 255, it should fail at much more places than just the tuplets

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**